pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -177,20 +178,33 @@ func (c *Reconciler) reconcile(ctx context.Context, pa *pav1alpha1.PodAutoscaler
 	// Compare the desired and observed resources to determine our situation.
 	// We fetch private endpoints here, since for scaling we're interested in the actual
 	// state of the deployment.
-	got := 0
+	ready, notReady, pending, terminating := 0, 0, 0, 0
 
 	// Propagate service name.
 	pa.Status.ServiceName = sks.Status.ServiceName
 	// Currently, SKS.IsReady==True when revision has >0 ready pods.
 	if sks.Status.IsReady() {
-		podCounter := resourceutil.NewScopedEndpointsCounter(c.endpointsLister, pa.Namespace, sks.Status.PrivateServiceName)
-		got, err = podCounter.ReadyCount()
+		podEndpointCounter := resourceutil.NewScopedEndpointsCounter(c.endpointsLister, pa.Namespace, sks.Status.PrivateServiceName)
+		ready, err = podEndpointCounter.ReadyCount()
+		if err != nil {
+			return fmt.Errorf("error checking endpoints %s: %w", sks.Status.PrivateServiceName, err)
+		}
+
+		notReady, err = podEndpointCounter.NotReadyCount()
 		if err != nil {
 			return fmt.Errorf("error checking endpoints %s: %w", sks.Status.PrivateServiceName, err)
 		}
 	}
-	logger.Infof("PA scale got=%d, want=%d, ebc=%d", got, want, decider.Status.ExcessBurstCapacity)
-	return computeStatus(pa, want, got)
+
+	podCounter := resourceutil.NewNotRunningPodsCounter(c.podsLister, pa.Namespace, sks.Status.ServiceName)
+	err = podCounter.Get()
+	if err != nil {
+		return fmt.Errorf("error checking pods %s: %w", sks.Status.PrivateServiceName, err)
+	}
+	pending, terminating = podCounter.PendingCount(), podCounter.TerminatingCount()
+
+	logger.Infof("PA scale got=%d, want=%d, ebc=%d", ready, want, decider.Status.ExcessBurstCapacity)
```

Comment:
log the new integers as well?

Reply:
will log it as a separate log line to avoid cluttering the scaling information.